### PR TITLE
Add database methods for has any or all validators

### PR DESCRIPTION
**Pull request by prestonvanloon**
In an effort to split up #2069, I have added the database methods to query whether some or all validators exist in the database by their public key.

### CODE REVIEW

File: beacon-chain/db/validator.go [hunk 1]
```diff
@@ -78,10 +78,42 @@ func (db *BeaconDB) HasValidator(pubKey []byte) bool {
...
+		bkt := tx.Bucket(validatorBucket)
 
-		exists = a.Get(h[:]) != nil
+		exists = bkt.Get(h[:]) != nil
 		return nil
 	})
 	return exists
 }
+
+// HasAllValidators returns true if all validators in a list of public keys
```
Comment (0xKiwi) [inline beacon-chain/db/validator.go]: If all validators are in*

File: beacon-chain/db/validator.go [hunk 2]
```diff
@@ -78,10 +78,42 @@ func (db *BeaconDB) HasValidator(pubKey []byte) bool {
...
+// are in the bucket.
+func (db *BeaconDB) HasAllValidators(pubKeys [][]byte) bool {
+	return db.hasValidators(pubKeys, true /* requireAll */)
+}
+
+// HasAnyValidators returns true if any validator in a list of public keys
```
Comment (0xKiwi) [inline beacon-chain/db/validator.go]: Any validators*

File: beacon-chain/db/validator.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/db/validator.go]: I think it was correct before.

File: beacon-chain/db/validator.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/db/validator.go]: Likewise, this was already correct

File: beacon-chain/db/validator.go (hunk 1 above)
Comment (0xKiwi) [reply beacon-chain/db/validator.go]: Ah my bad I didn't see the second line!

**Pull request merged**